packages/ui/Input/Select.tsx

```diff
@@ -67,69 +87,167 @@ function Select<T extends string | number | boolean>(props: SelectProps<T>) {
+  children: React.ReactNode;
+}
+
+function SelectTrigger({ children }: SelectTriggerProps) {
+  const { open, setOpen, buttonRef } = useSelectContext();
+
+  const handleClick = () => {
+    setOpen(!open);
+  };
+
+  return (
+    <button className={S.buttonWithNoStyle} onClick={handleClick} ref={buttonRef} type='button'>
+      {children}
+    </button>
+  );
+}
+
+SelectTrigger.displayName = 'Select.Trigger';
+
+interface SelectTriggerContentProps {
+  className?: string;
+  placeholder?: string;
+}
+
+// Select.TriggerContent 컴포넌트: trigger의 미리 정의된 UI
+function SelectTriggerContent({ className, placeholder }: SelectTriggerContentProps) {
+  const { open, selected } = useSelectContext();
+
+  const selectedLabel = selected ? selected.label : placeholder;
+
+  return (
+    <div className={`${S.select} ${className ? className : ''}`}>
+      <p className={!selected ? S.selectPlaceholder : ''}>{selectedLabel}</p>
+      <IconChevronDown
+        style={{
+          width: 20,
+          height: 20,
+          transform: open ? 'rotate(-180deg)' : '',
+          transition: 'all 0.3s ease',
+        }}
+      />
     </div>
   );
 }
 
+SelectTriggerContent.displayName = 'Select.TriggerContent';
+
+interface SelectMenuProps {
+  children: React.ReactNode;
+}
+
+// SelectMenu 컴포넌트: 옵션 목록을 렌더링
+function SelectMenu({ children }: SelectMenuProps) {
+  const { open, optionsRef, calcMaxHeight } = useSelectContext();
+
+  if (!open) {
+    return null;
+  }
+
+  return (
+    <ul className={S.optionList} ref={optionsRef} style={{ maxHeight: calcMaxHeight() }}>
+      {children}
+    </ul>
+  );
+}
+
+SelectMenu.displayName = 'Select.Menu';
+
+interface SelectMenuItemProps<T> {
+  option: Option<T>;
+  onClick?: () => void;
+}
+
+// SelectMenuItem 컴포넌트: 옵션 목록 하나의 UI
+function SelectMenuItem<T>({ option, onClick }: SelectMenuItemProps<T>) {
+  const { open, type, handleOptionClick } = useSelectContext();
+
+  const handleClick = () => {
+    handleOptionClick(option);
+
+    if (onClick) {
+      onClick();
+    }
+  };
+
+  if (!open) {
+    return null;
+  }
+
+  return (
+    <li>
+      <button className={S.option} onClick={handleClick} type='button'>
+        {type === 'textIcon' && option.icon}
+        {(type === 'userList' || type === 'userListDesc') &&
+          (option.profileUrl ? (
+            <img alt={option.label} className={S.optionProfileImg} src={option.profileUrl} />
+          ) : (
+            <div className={S.optionProfileEmpty}>
+              <IconUser />
+            </div>
+          ))}
+
+        <div>
+          <p>{option.label}</p>
+          {(type === 'textDesc' || type === 'userListDesc') && <p className={S.optionDesc}>{option.description}</p>}
+        </div>
+      </button>
+    </li>
+  );
+}
+
+SelectMenuItem.displayName = 'Select.MenuItem';
+
+const Select = {
+  Root: SelectRoot,
+  Trigger: SelectTrigger,
+  TriggerContent: SelectTriggerContent,
+  Menu: SelectMenu,
+  MenuItem: SelectMenuItem,
+};
```

Comment:
```suggestion
const Select = {
  Root,
  Trigger,
  TriggerContent,
  Menu,
  Select.MenuItem: MenuItem
};
```

이렇게 컴포넌트명을 단순화해도 될 것 같다는 생각이네요! Select  컴포넌트에 들어있는 컴포넌트들이니 Select라는 이름을 중복해서 사용할 필요는 없다는 생각입니다.

Reply:
저는 이 부분을 Compound Component 패턴에서 서브컴포넌트들을 그룹핑한 코드로 이해했어요.
내부적으로 구현한 SelectRoot 컴포넌트를 실제 구현 레벨에서 <Select.Root>처럼 사용할 수 있도록 해준 부분으로 보이는데 제가 잘못 이해한 걸까요...? 아니면 혹시 내부적인 SelectRoot 컴포넌트를 Root라는 네이밍으로 변경하는 게 좋을 것 같다는 말씀이신지 궁금합니다!!

Reply:
> 혹시 내부적인 SelectRoot 컴포넌트를 Root라는 네이밍으로 변경하는 게 좋을 것 같다는 말씀이신지 궁금합니다!!

이겁니다! 외부에서는 어차피 Select.*으로 묶여 제공되기 때문에 내부 네이밍에까지 Select라는 이름이 중복으로 존재하지 않아도 될 것 같아 제안드렸습니다.

Reply:
MDS `Dialog` 컴포넌트도 합성 컴포넌트 패턴으로 구성되어 있는데, 서브 컴포넌트들의 네이밍을 다음과 같이 했더라구요.

```tsx
export function DialogTitle({ children }: ChildrenProp) {
  return <div className={title}>{children}</div>;
}

export function DialogDescription({ children }: DialogDescriptionProps) {
  return <div className={description}>{children}</div>;
}

export function DialogFooter({ children, align }: DialogFooterProp) {
  return <div className={`${footer[align]} ${gap}`}>{children}</div>;
}

export function DialogAnimation({ children }: ChildrenProp) {
  return <div className={animation}>{children}</div>;
}

```

만약 제안 주신것과 같이 수정한다면 통일성을 위해 이 부분도 수정하는 게 좋아보입니다!!

Reply:
흠 그렇네요... 이건 단순히 컨벤션 문제일 것 같은데, 어느 쪽이 더 괜찮아 보이시나요? Select* 로 간다면

중간중간 컴포넌트에 있는 displayName 만 제거해줘도 될 듯 합니다.

Reply:
@Brokyeom @jungwoo3490 , 이 부분은 일관성있게만 가져간다면 문제가 없을것 같습니다. 저는 개인적으로 이런 합성 컴포넌트 방식으로 컴포넌트를 가져가게 된다면, Select* 형식으로 이름을 붙이는게 어떤 컴포넌트인지 구분하기 편하다고 생각되는데 어떠신가요?!

- displayName은 추후 디버깅에 좋다고 해서 넣어두었는데, 이 부분은 필요성을 느낄 때 사용하는게 더 의미 있을것 같아서 제거하겠습니다.

Reply:
Select* 로 가는 것도 좋아보입니다!! 가능성이야 희박하겠지만 네임스페이스 충돌도 방지할 수 있고, 빠르게 검색해서 디버깅하기도 좋을 것 같아요!!
forwardRef를 사용하는 컴포넌트에서 디버깅이 힘들다는 문제점 때문에 displayName을 많이 사용하는 걸로 알고 있는데, 지금은 필요없을 것 같다는 생각도 듭니다!!

Reply:
> Select* 로 가는 것도 좋아보입니다!!

이걸로 가시죠!